mitodl/mitxonline
# Catalog Update without having to hit bottom of page

### What are the relevant tickets?
fixes https://github.com/mitodl/hq/issues/3141
fixes https://github.com/mitodl/hq/issues/2992
fixes https://github.com/mitodl/hq/issues/2910
fixes https://github.com/mitodl/hq/issues/2909

### Description (What does it do?)
Rather than relying on the page scroll to load more, there is a chain of requests to fill out the data for the page until it is full, regardless of page movement, once the initial load is complete.

### How can this be tested?
Set up data with more than 12 courses and/or programs (ideally both). You will want a few departments loaded as well.  Load the catalog and click to filter the list.  You should get all cards loading in.
